umpleonline/scripts/umple_action.js

```diff
@@ -1571,6 +1590,63 @@ Action.updateUmpleDiagramCallback = function(response)
         // jQuery("div.umpleClass").addClass("unselectable");
       }
     }
+    else if(Page.useJointJSClassDiagram) {
+      // console.log(diagramCode);
+      var model = JSON.parse(diagramCode);
+
+      var umpleCanvas = jQuery("#umpleCanvas");
+      umpleCanvas.html('<div id="myholder"></div>');
+
+      var graph = new joint.dia.Graph;
+
+      // start my making the paper the same size as umpleCanvas and then scaling the model to fit
+      var paper = new joint.dia.Paper({
+          el: jQuery("#myholder"),
+          width: umpleCanvas.width(),
+          height: umpleCanvas.height(),
+          model: graph,
+          gridSize: 1
+      });
+
+      graph.addCells(JJsParse.makeClasses(model));
+      graph.addCells(JJsParse.makeAssociations(model));
+
+      // call this after having added all diagram elements to scale them to fit the available space
+      paper.scaleContentToFit({padding: 15});
+
+      // zooming with the mouse wheel or finger swipe
+      var MouseWheelHandler = function (event){
+        // console.log(event.deltaY);
+        if (event.ctrlKey === true) {
+          var paperHeight = paper.options.height;
+          var paperWidth = paper.options.width;
+          // scaleFactor is either 1.1 or 0.9
+          var scaleFactor = 1 + (Math.abs(event.deltaY) / (event.deltaY * 10));
+          // console.log(scaleFactor);
+          paper.setDimensions(paperWidth * scaleFactor, paperHeight * scaleFactor)
+          // paper.height = paperHeight * scaleFactor;
+          // paper.width = paperWidth * scaleFactor;
+
+          // console.log(paper);
+          console.log(paper.options);
```

Comment:
Shouldn't leave console logging in a "production" environment, same with commented code, unless its relevant.